### ISSUE TITLE
Unexpected value for Arms Warrior Strength weight

### ISSUE THREAD

**Issue by jdelvare** [wowsims/wotlk#3154]
**Describe the bug**
When computing the stat weights for my Arms Warrior, I get very different and unexpected weight values for the Strength stat.

**To Reproduce**
1. Open an Arms Warrior simulation and set the RNG seed to 42 and iterations to 6000.
2. Run the Stats Weights.
3. Write down the Strength weight.
4. Make an arbitrary change to the character's stats, for example remove 30 melee hit score.
5. Run the Stats Weights again.
6. Write down the new Strength weight.
7. Compare both values. In my case, it changed from 2.11 to 2.51.

**Expected behavior**
Considering that the equivalence between Strength and Attack Power can be accurately computed, and that variations in the Strength value, just like variations in Attack Power values, can't change the sequence of events during a fixed RNG simulation, I would expect the Stats Weights computation to always display the same, expected weight for Strength.

With Benediction of Kings and the [Strength of Arms](https://wowclassicdb.com/wotlk/spell/46866) talent, one point of Strength gives exactly 2 * 1.10 * 1.04 = 2.29 AP. This is the value I expect from any Stats Weights run.

Note that my expectation about Strength weight is met in the Feral Druid simulator. For a Feral Druid with Benediction of Kings and the usual talents, the weight of Strength should be 2 * 1.10 * 1.02 * 1.06 = 2.38 and this is exactly the value being returned by a Stats Weights run.

So I don't understand why this works as expected for the Feral Druid and not for the Arms Warrior. Am I missing some characteristics or talent of the Warrior which causes Strength to change more than the Attack Power?

**Desktop (please complete the following information):**
 - OS: openSUSE Leap 15.4 (Linux)
 - Browser: Firefox 102.11.0esr (64-bits)

**Additional context**
An interesting experiment I made is to simulate the Strength value manually. Still using a fixed RNG value of 42 and 6000 iterations, I tested the effect of manually adding 100 Attack Power. Result: +47.50 DPS. Then I restored the original Attack Power value and manually added 100 Strength. Result +116.40 DPS. That's a ratio of 2.45, not the expected 2.29.

If I perform the very same test on my Feral Druid, the ratio is 2.38, which is the expected value.

So there's definitely something odd about the Warrior's Strength. Either a bug, or a mechanism I do not understand.

**Comment by TheGroxEmpire** [comment]
Comparing Feral cat and warrior in terms of EP weights is not ideal. Warriors gain rage from damage done which will affect the EP. The EP value for warrior takes a lot more iteration than feral cat to get consistent value due to this. If you do horrendous amount of iteration per weights, you'll get much closer value.

I did this with same seed 42 and default sim on P3 bis with 100000 iterations:
Reference:
![image](https://github.com/wowsims/wotlk/assets/47816228/28d73149-8aee-4b80-ab05-49e73c432c22)

-130 hit ratings:
![image](https://github.com/wowsims/wotlk/assets/47816228/40290707-b7a4-423b-aa3d-2cfd7a1189ce)

**Comment by jdelvare** [comment]
Thanks for pointing out the rage factor, that's what I was missing. I assumed we got a fixed amount of rage for hitting, I did not expect the amount of rage we get to depend on the actual damage done. Out of curiosity, is the damage-to-rage formula available somewhere?

As a consequence, my assumption that changing the AP or Strength did not change the sequence of events when the RNG is set to a fixed value, was incorrect for the Warrior (while it still holds for the Feral Druid in Cat form) and this indeed explains why a limited number of iterations will lead to a lot of noise when computing Stats Weights (and not just for Strength actually).

I followed your advice and ran my manual tests again, this time with 1,000,000 iterations. I got +50.80 DPS for +100 AP and +116.31 for +100 Strength, that's a ratio of 2.29 which matches the theoretical value exactly. Problem solved, thank you very much.
